tests/Struct/FindologicProductTest.php

```diff
@@ -333,20 +348,29 @@ public function testProductPrices(?Price $price, ?string $exception): void
         }
     }
 
+    public function directIntegrationProvider()
+    {
+        return [
+            'Direct Integration' => [true],
+            'API' => [false],
```

Comment:
```suggestion
            'Integration type is Direct Integration' => [
                'integrationType' => 'Direct Integration'
            ],
            'Integration type is API' => [
                'integrationType' => 'API'
            ],
```

Reply:
Please add pre-defined results instead of dynamically getting the required result. This may be done by creating a separate test with a data provider. This is what the test cases in the story description were designed for E.g.

```php
public function categoryAndCatUrlWithIntegrationTypeProvider(): array
{
    return [
        'Integration type is API and category is at first level' => [
             'integrationType' => 'API',
             'categories' => [
                 'id-of-assigned-category'
             ],
             'expectedCategories' => [
                'SomeCategoryName'
             ],
             'expectedCatUrls' => [],
        ],
        'Integration type is API with nested categories' => [
             'integrationType' => 'API',
             'categories' => [
                 'id-of-assigned-category'
             ],
             'expectedCategories' => [
                'Some_Category_Path' // Not recursive
             ],
             'expectedCatUrls' => [],
        ],
        'Integration type is DI and category is at first level' => [
             'integrationType' => 'Direct Integration',
             'categories' => [
                 'id-of-assigned-category'
             ],
             'expectedCategories' => [
                'SomeCategoryName'
             ],
             'expectedCatUrls' => [
                'SomeCategoryPath'
             ],
        ],
        'Integration type is DI with nested categories' => [
             'integrationType' => 'Direct Integration',
             'categories' => [
                 'id-of-assigned-category'
             ],
             'expectedCategories' => [
                'Some_Category_Path',
                'Some_Category',
                'Some',
             ],
             'expectedCatUrls' => [
                'Some/Category/Path',
                'Some/Category',
                'Some',
             ],
        ],
        'Integration type is unknown and category is at first level' => [
             'integrationType' => null,
             'categories' => [
                 'id-of-assigned-category'
             ],
             'expectedCategories' => [
                'SomeCategoryName'
             ],
             'expectedCatUrls' => [
                'SomeCategoryPath'
             ],
        ],
        'Integration type is unknown with nested categories' => [
             'integrationType' => null,
             'categories' => [
                 'id-of-assigned-category'
             ],
             'expectedCategories' => [
                'Some_Category_Path',
                'Some_Category',
                'Some',
             ],
             'expectedCatUrls' => [
                'Some/Category/Path',
                'Some/Category',
                'Some',
             ],
        ],
    ];
}

public function integrationTypeSpecialCharacterProvider(): array
{
    return [
        'Integration type is API' => [
            'integrationType' => 'API',
            'attributeValue' => '<><<<##test',
            'expectedAttributeValue' => 'test',
        ],
        'Integration type is DI' => [
            'integrationType' => 'Direct Integration',
            'attributeValue' => '<><<<##test',
            'expectedAttributeValue' => '<><<<##test',
        ],
        'Integration type is unknown' => [
            'integrationType' => 'Direct Integration',
            'attributeValue' => '<><<<##test',
            'expectedAttributeValue' => 'test',
        ],
    ];
}
```